src/storage-adapters/mongo/index.test.js

```diff
@@ -0,0 +1,501 @@
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+import chai from 'chai';
+import config from 'config';
+import { MongoClient } from 'mongodb';
+
+import MongoAdapter from './index.js';
+
+const { expect } = chai;
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+const { connectionURI } = config.get('recorder.snapshots.storage.mongo');
+const client = new MongoClient(connectionURI);
+
+const SERVICE_PROVIDER_ID = 'test_service';
+const DOCUMENT_TYPE = 'Terms of Service';
+const CONTENT = 'ToS fixture data with UTF-8 çhãràčtęrs';
+const MIME_TYPE = 'text/html';
+const FETCH_DATE = new Date('2000-01-01T12:00:00.000Z');
+const FETCH_DATE_LATER = new Date('2000-01-02T12:00:00.000Z');
+const SNAPSHOT_ID = '61af86dc5ff5caa74ae926ad';
+const PDF_CONTENT = fs.readFileSync(path.resolve(__dirname, '../../../test/fixtures/terms.pdf'), { encoding: 'utf8' });
+const UPDATED_PDF_CONTENT = fs.readFileSync(path.resolve(__dirname, '../../../test/fixtures/termsModified.pdf'), { encoding: 'utf8' });
+const PDF_MIME_TYPE = 'application/pdf';
+
+let collection;
+
+describe('MongoAdapter', () => {
+  let subject;
+
+  before(async () => {
+    subject = new MongoAdapter(config.get('recorder.snapshots.storage.mongo'));
+    await subject.initialize();
+    await client.connect();
+    const db = client.db(config.get('recorder.snapshots.storage.mongo').database);
+
+    collection = db.collection(config.get('recorder.snapshots.storage.mongo').collection);
+  });
+
+  describe('#record', () => {
+    let record;
+    let mongoDocument;
+    let numberOfRecordsBefore;
+    let numberOfRecordsAfter;
+
+    context('when it is the first record', () => {
+      before(async () => {
+        numberOfRecordsBefore = await collection.find({
+          serviceId: SERVICE_PROVIDER_ID,
+          documentType: DOCUMENT_TYPE,
+        }).count();
+
+        (record = await subject.record({
+          serviceId: SERVICE_PROVIDER_ID,
+          documentType: DOCUMENT_TYPE,
+          content: CONTENT,
+          mimeType: MIME_TYPE,
+          fetchDate: FETCH_DATE,
+          snapshotId: SNAPSHOT_ID,
+        }));
+
+        numberOfRecordsAfter = await collection.find({
+          serviceId: SERVICE_PROVIDER_ID,
+          documentType: DOCUMENT_TYPE,
+        }).count();
+
+        (mongoDocument = await collection.findOne({
+          serviceId: SERVICE_PROVIDER_ID,
+          documentType: DOCUMENT_TYPE,
+        }));
+      });
+
+      after(() => collection.drop());
```

Comment:
maybe `after(async () => subject._removeAllRecords());`

To be consistent with GIT adapter